external/llvm-project/mlir/include/mlir/Dialect/Tosa/Transforms/Passes.h

```diff
@@ -31,9 +31,9 @@ std::unique_ptr<Pass> createTosaInferShapesPass();
 std::unique_ptr<Pass> createTosaMakeBroadcastablePass();
 std::unique_ptr<Pass> createTosaTestQuantUtilAPIPass();
 std::unique_ptr<Pass> createTosaPartitionPass();
-std::unique_ptr<Pass> createPostPartitionCollapsePass();
+class PartitionConfig;
+std::unique_ptr<Pass> createTosaPartitionPass(PartitionConfig *config);
```

Comment:
Why's this by-pointer?

Reply:
Maybe I have a blind spot, but I'm not aware that you can make virtual calls on things that aren't pointers.

Reply:
Yeah, you can make virtual calls on references (which are a very funky way to spell non-null pointers, except when they aren't)